### PR TITLE
Multi color component

### PR DESCRIPTION
**Pull request by evoinea**
Contributes to https://github.com/IPG-Automotive-UK/instrument-designer/issues/120

This pull request:
* Adds new MultiColor component.
* Updates Color component: reduce re-renders by reducing amount of local state and adding memo
* Debounces Color callback to fix case ending in infinite re-renders  
* Installs latest @mui/materials library

image

### CODE REVIEW

File: src/MultiColor/MultiColor.test.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  return Array.from(
+    document.querySelectorAll(`[role="cell"][data-colindex="${colIndex}"]`)
+  ).map(node => node.textContent);
+}
+
+/**
+ * Tests
+ */
+describe("MultiColor", () => {
+  test("can select an item", () => {
```
Comment (dmbarry86) [inline src/MultiColor/MultiColor.test.js:38]: Are you sure this is a test for 'can select an item'? I don't see any selection going on. Looks like you are just rendering MultiColor with some data and checking the value column is as expected. Good test, bad name. You might also want to check the color column appears as expected too.

File: src/MultiColor/MultiColor.test.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    render(<MultiColorWithState />);
+    expect(Number(getColumnValues(0))).toBe(2);
+  });
+  test("can update an item", () => {
```
Comment (dmbarry86) [inline src/MultiColor/MultiColor.test.js:42]: updating value not item - there is no item

File: src/MultiColor/MultiColor.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+import Color from "../Color";
+import { DataGrid } from "@mui/x-data-grid";
+import DeleteIcon from "@mui/icons-material/Delete";
+import PropTypes from "prop-types";
+import React from "react";
+
+/**
+ * Multi color component is used to manage a table for value-color pair
+ */
+
```
Comment (dmbarry86) [inline src/MultiColor/MultiColor.js:12]: remove space here

File: src/MultiColor/MultiColor.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+export default function MultiColor({
+  onChange = () => {},
+  rows = [],
+  style = { height: 250, width: "100%" }
+}) {
+  // add an id for each color/value
+  const rowsWithID = rows.map((item, index) => ({ ...item, id: index }));
+  // set column definition
```
Comment (dmbarry86) [inline src/MultiColor/MultiColor.js:20]: add newline before this

File: src/MultiColor/MultiColor.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+   */
+  onChange: PropTypes.func,
+  /**
+   * Set of rows.
+   */
+  rows: PropTypes.array,
+  /**
+   * Custom style to apply to the component.
+   */
+  style: PropTypes.object
```
Comment (dmbarry86) [inline src/MultiColor/MultiColor.js:139]: What is the plan for this style prop? Why is it needed?

File: src/MultiColor/MultiColor.test.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  const handleChange = newRows => {
+    setRows(newRows);
+    onChange && onChange(newRows);
+  };
+  return (
+    <MultiColor {...rest} autoHeight onChange={handleChange} rows={rows} />
+  );
+};
+
+function getColumnValues(colIndex) {
```
Comment (dmbarry86) [inline src/MultiColor/MultiColor.test.js:28]: add comment to explain what this function is doing

**Review by jegasriskantha — APPROVED** [review]
LGTM

**Pull request merged**